omnipaxos/src/sequence_paxos/mod.rs

```diff
@@ -246,13 +245,18 @@ where
         }
     }
 
-    /// Returns the outgoing messages from this replica. The messages should then be sent via the network implementation.
-    pub(crate) fn get_outgoing_msgs(&mut self) -> Vec<PaxosMessage<T>> {
-        let mut outgoing = Vec::with_capacity(self.buffer_size);
-        std::mem::swap(&mut self.outgoing, &mut outgoing);
+    /// Moves the outgoing messages from this replica into the buffer. The messages should then be sent via the network implementation.
+    /// If the buffer is empty it can be efficently swapped, otherwise messages must be copied into
```

Comment:
```suggestion
    /// If `buffer` is empty, it gets swapped with the internal message buffer. Otherwise, messages are appended to the buffer. This prevents messages from getting discarded.
```